docs/markdown/Commands.md

```diff
@@ -340,15 +340,6 @@ These variables are set in environment in addition to those set using `meson.add
   schemas is compiled. This is automatically set when using `gnome.compile_schemas()`.
   Note that this requires GLib >= 2.64 when `gnome.compile_schemas()` is used in
   more than one directory.
-- `PYTHONPATH` *Since 0.62.0* includes every directory where a python module is being
-  installed using [`python.install_sources()`](Python-module.md#install_sources)
-  and [`python.extension_module()`](Python-module.md#extension_module). Python
-  modules installed by other means, such as `install_data()` or `install_subdir()`,
-  will not be included and should be added to `PYTHONPATH` manually using
-  [`meson.add_devenv()`](Reference-manual_builtin_meson.md#mesonadd_devenv).
-  Note that when modules are installed into subdirectories the source tree
-  layout must match the installed tree layout otherwise `import subdir.mod`
-  cannot work.
```

Comment:
Agreed, this should be removed. Use of PYTHONPATH is not a habit that tools should teach to users even if it would work reliably, which it definitely won't for out of tree builds.

Long-term we can indeed make editable installs (`pip install . -e` and the like) work.

Reply:
> Use of PYTHONPATH is not a habit that tools should teach to users even if it would work reliably

Why?

I think it's still useful even if it has limitations. devenv has plenty of limitations, it's mostly a best effort to help developers when possible. Can of course always be extended by symlinking everything in a common location, can be done later IMHO.

My use-case when I wrote this is a python app that has a C library that gets used by python with gobject-introspection. devenv already sets GI_TYPELIB_PATH so python bindings for glib based libraries does work.

Reply:
I disagree with the idea that we should expect limitations and implement broken stuff that actively makes things worse, because "it's just a best effort".

Instead, we should say that devenv doesn't implement everything, so yes it is limited, but what it does implement is always better than not implementing it.

Using highly loaded language like "it's just a best effort" is not needed and has no effect here other than to play a sympathy card which I don't think is topical. We can evaluate each component of devenv based on whether it is individually productive, removing PYTHONPATH isn't an indictment on the rest of devenv, and we don't need to make excuses for why a destructive component should stay in regardless because "it has good intentions".

Reply:
In fact, as mentioned in the commit message, devenv really is super useful! It provides an easy, helpful framework for extending itself via `meson.add_devenv()` for when you as the developer of a specific project know that PYTHONPATH will actually work correctly in that project.

I am very much in support of devenv! Used properly. Where we only implement stuff expected to be correct, and provide people easy tools to build on that in project-specific ways.

Reply:
@xclaesse "why?" can have a long or a very short answer. The long one involves years of history on PYTHONPATH, how `setuptools` didn't treat it right, env vars not being best practice, etc. But the short one is probably enough: the consensus in the Python packaging crowd is that it's not the thing to teach or use. For valid use cases there should be better designed alternatives - which in this case is editable installs.

Note that I personally do use PYTHONPATH on occasion, it can be handy. But it's not something that should be taught or made the default in a widely used tool like Meson.

Reply:
> But the short one is probably enough: the consensus in the Python packaging crowd is that it's not the thing to teach or use. 

Based on this it seems like a reasonable thing to revert this.